### PR TITLE
(#143) Fix title- vs. lower-case in fallback logic

### PR DESCRIPTION
**Pull request by sarinapadilla**
* Update fallback logic to lowercase dictionary param
* Update fallback combinations to use lowercase dictionary param
* Update GetById tests for lower- and title-case params
* Update GetByName tests for lower- and title-case params
* Update integration tests to account for lower- and title-case params

### CODE REVIEW

File: test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.GetByName.cs [hunk 1]
```diff
@@ -360,7 +360,7 @@ public async void GetByName_WithFallback_TermsPatient()
             .Returns(Task.FromResult(glossaryTerm));
 
             TermsController controller = new TermsController(NullLogger<TermsController>.Instance, termQueryService.Object);
-            GlossaryTerm gsTerm = await controller.GetByName("NotSet", AudienceType.Patient, "en", "s-phase-fraction", true);
+            GlossaryTerm gsTerm = await controller.GetByName("notset", AudienceType.Patient, "en", "s-phase-fraction", true);
```
Comment (blairlearn) [inline test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.GetByName.cs:363]: Leaving this as mixed-case `NotSet` would be helpful for catching problems if the controller stops handling it correctly.  Ideally, that would be detected by tests for that specific purpose, but leaving this the way it was would not have been bad.

File: test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.cs [hunk 2]
```diff
@@ -431,7 +431,7 @@ public async void GetById_WithFallback_TermsPatient()
             .Returns(Task.FromResult(glossaryTerm));
 
             TermsController controller = new TermsController(NullLogger<TermsController>.Instance, termQueryService.Object);
-            GlossaryTerm gsTerm = await controller.GetById("NotSet", AudienceType.Patient, "en", 44771, true);
+            GlossaryTerm gsTerm = await controller.GetById("notset", AudienceType.Patient, "en", 44771, true);
```
Comment (blairlearn) [inline test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.cs:434]: Again, not a _bad_ change, just not  a necessary one.

**Review by blairlearn — CHANGES_REQUESTED** [review]
The file `integration-tests/features/terms/getById/success.feature` (formerly `integration-tests/features/getById.feature`) contains no actual tests and should be deleted.

**Comment by blairlearn** [comment]
@sarinapadilla and I have discovered that the Elasticsearch mapping for the `dictionary` field needs to be changed to use the `caseinsensitive_normalizer`.

```json
                "dictionary": {
                    "type": "keyword",
                    "normalizer": "caseinsensitive_normalizer"
                },
```

@bkline, please make this change in the glossary loader and push it to the DEV tier.

File: src/NCI.OCPL.Api.Glossary/Controllers/TermsController.cs [hunk 3]
```diff
@@ -75,14 +75,24 @@ public async Task<GlossaryTerm> GetById(string dictionary, AudienceType audience
...
+            // If none of the options return a term, then throw an error.
             else {
                 // Set order of fallback options based on current audience.
                 var fallbackOptions = audience == AudienceType.Patient ? _fallbackOptionsPatient : _fallbackOptionsHP;
 
-                Tuple<string, AudienceType> currentValue = new Tuple<string, AudienceType>(dictionary, audience);
-                LinkedListNode<Tuple<string, AudienceType>> start = fallbackOptions.Find(currentValue);
+                Tuple<string, AudienceType> requestedOptions = new Tuple<string, AudienceType>(dictionary.ToLower(), audience);
+                Tuple<string, AudienceType> initialOption = fallbackOptions.FirstOrDefault(t => (t.Item1.ToLower() == requestedOptions.Item1) && (t.Item2 == requestedOptions.Item2) );
+                LinkedListNode<Tuple<string, AudienceType>> start = fallbackOptions.Find(initialOption);
```
[the remark this answers is not in the document]
Comment (sarinapadilla) [reply src/NCI.OCPL.Api.Glossary/Controllers/TermsController.cs]: Updated according to your comment - the fallbackOptions all have lowercase dictionary names that are set from class-level constants, and the dictionary argument is lowercased. This is updated for both GetById and GetByName.

The unit tests are updated accordingly (calls to the query service when useFallback=true all use lowercase dictionary names). Integration tests are now passing thanks to using the caseinsensitive normalizer for the dictionary (which will be updated in the glossary loader).

File: test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.GetByName.cs (hunk 1 above)
Comment (sarinapadilla) [reply test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.GetByName.cs]: Removed, as passing in the title case is a good test to have (even though we test it all over this test file).

File: test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.cs (hunk 2 above)
Comment (sarinapadilla) [reply test/NCI.OCPL.Api.Glossary.Tests/Tests/Controllers/TermsControllerTest.cs]: Same answer as above!

**Comment by bkline** [comment]
I have reloaded ncias-d2433-v using the requested modification to the mapping file.

**Review by blairlearn — APPROVED** [review]

**Pull request merged**